src/services/Store/Account/useAccounts.tsx

```diff
@@ -54,14 +56,19 @@ function useAccounts() {
     actionName: 'Tx Made'
   });
 
-  const createAccountWithID = (uuid: TUuid, item: IRawAccount) => {
-    addAccountToFavorites(uuid);
-    dispatch(createAccount({ ...item, uuid }));
+  const createAccountWithID = (uuid: TUuid, item: IRawAccount, isDemoMode?: boolean) => {
+    addAccountToFavorites(uuid, isDemoMode);
+    dispatch(
+      isDemoMode ? resetAndCreateAccount({ ...item, uuid }) : createAccount({ ...item, uuid })
+    );
   };
 
-  const createMultipleAccountsWithIDs = (newAccounts: IAccount[]) => {
-    addMultipleAccountsToFavorites(newAccounts.map(({ uuid }) => uuid));
-    dispatch(createAccounts(newAccounts));
+  const createMultipleAccountsWithIDs = (newAccounts: IAccount[], isDemoMode?: boolean) => {
+    addMultipleAccountsToFavorites(
+      newAccounts.map(({ uuid }) => uuid),
+      isDemoMode
+    );
+    dispatch(isDemoMode ? resetAndCreateManyAccounts(newAccounts) : createAccounts(newAccounts));
```

Comment:
We might need to move some of this to the reducer, as we will be deleting the hooks at one point. Perhaps it could be a saga that either triggers `resetAndCreateManyAccounts` or `createAccounts`. See the network slice for a similar pattern!